onnx/onnx-ml.proto3

```diff
@@ -242,20 +242,28 @@ message TrainingInfoProto {
   // iteration to zero.
   //
   // By default, this field is an empty graph and its evaluation does not
-  // produce any output.
+  // produce any output. Thus, no initializer would be changed by default.
   GraphProto initialization = 1;
 
   // This field represents a training algorithm step. Given required inputs,
   // it computes outputs to update initializers in its own or inference graph's
-  // initializer lists. In general, this graph contains loss node, gradient node,
-  // optimizer node, increment of iteration count, and some calls to the inference
-  // graph.
+  // initializer lists. In general, this field contains loss node, gradient node,
+  // optimizer node, increment of iteration count.
   //
-  // The field algorithm.node is the only place the user can use GraphCall
-  // operator. The only callable graph is the one stored in ModelProto.graph.
+  // An execution of the training algorithm step is performed by executing the
+  // graph obtained by combining the inference graph (namely "ModelProto.graph")
+  // and the "algorithm" graph. That is, the actual the actual
+  // input/initializer/output/node/value_info/sparse_initializer list of
+  // the training graph is the concatenation of
+  // "ModelProto.graph.input/initializer/output/node/value_info/sparse_initializer"
+  // and "algorithm.input/initializer/output/node/value_info/sparse_initializer"
+  // in that order. This combined graph must satisfy the normal ONNX conditions.
+  // For example, an input of a node in the "algorithm" graph may reference the
+  // output of a node in the inference graph (but not the other way round).
```

Comment:
Maybe add some more comments.

Inf:
	ModelProto.grpah
		initializer_w -> MatMul1 -> tensor_x_inf

Algorithm:
	ModelProto.training_info[i].algorithm
		initializer_w -> GraphCall(MatMul1) -> tensor_x -> MatMul2 -> tensor_y -> MatMul3 -> tensor_z


Inf:
	ModelProto.grpah
                initializer_w -> MatMul1 -> tensor_x

Algorithm:
	ModelProto.training_info[i].algorithm
                       tensor_x -> MatMul2 -> tensor_y -> MatMul3 -> new_initializer_w
	ModelProto.training_info[i].update_binding
		       "initializer_w", "new_initializer_w"

Underlying graph represented by TrainingInfoProto.algorithm (with the context of inference graph)
 GraphProto.input = concat(ModelProto.input, algorithm.input)
 GraphProto.node = concat(ModelProto.node, algorithm.node)
 GraphProto.output = concat(ModelProto.output, algorithm.output)